File: src/v1/applications/post/logic.js
const {
  createApplication,
  emitApplicationCreated,
} = require('/opt/ports');

/**
 * Business logic
 * @returns {string} Application ID
 */

async function logic() {
  const applicationId = await createApplication();
  await emitApplicationCreated(applicationId);
  return applicationId;
}

module.exports = {
  logic,
};

File: src/cmd/createApplication/logic.go
package main

import (
	"github.com/thomasstep/authentication-service/internal/adapters"
)

func logic() (string, error) {
	applicationId, appErr := adapters.CreateApplication()
	if appErr != nil {
		return "", appErr
	}

	emitErr := adapters.EmitApplicationCreated(applicationId)
	if emitErr != nil {
		return "", emitErr
	}

	return applicationId, nil
}
